lib/slack/real_time/concurrency/celluloid.rb

```diff
@@ -38,6 +38,8 @@ def run_loop
           rescue EOFError => e
             logger.debug("#{self.class}##{__method__}") { e }
             driver.emit(:close, WebSocket::Driver::CloseEvent.new(1001, 'server closed connection')) unless @closing
+          ensure
+            terminate
```

Comment:
I will submit the PR with changes. I am testing a variation of the fix currently. 

PS: I reproduced the issue by running the code from ruby console. When you exit the console you will see the exception.